### ISSUE TITLE
Unable to reproduce main results from the paper

### ISSUE THREAD

**Issue by ZhouGengmo** [pgmikhael/clipzyme#5]
Thanks for your excellent work!

Unfortunately, I'm having trouble reproducing the main results of the paper, specifically the first row in Table 1. The expected results are:
BEDROC85: 44.69%
BEDROC20: 62.98%
EF0.05: 14.09
EF0.1: 8.06

**Environment Setup**
I followed the instructions from the [Installation](https://github.com/pgmikhael/clipzyme/tree/clipzyme?tab=readme-ov-file#installation) section of the README to set up the environment.

Here are some of the key package versions I used:
torch: 2.0.1+cu118
torch-cluster: 1.6.3+pt20cu118
torch-geometric: 2.3.1
torch-scatter: 2.1.2+pt20cu118
torch-sparse: 0.6.18+pt20cu118
torch-spline-conv: 1.2.2+pt20cu118
fair-esm: 2.0.0
rdkit: 2024.3.5

**Reproduction Attempt 1:**
I followed the reproduction steps in the [Training and Evaluation](https://github.com/pgmikhael/clipzyme/tree/clipzyme?tab=readme-ov-file#training-and-evaluation) section of the README.

I downloaded `esm2_t33_650M_UR50D` and set `esm_dir` correctly. And I encountered the following error when running the script:
```
python scripts/dispatcher.py -c configs/eval/clip_egnn.json -l ./logs/

FileNotFoundError: [Errno 2] No such file or directory: 'esm2_t33_650M_UR50D-contact-regression.pt'
```
I fail to find this file in the released data and models.

**Reproduction Attempt 2:**
I tried to use the trained model to infer on the test set based on the `scripts/Results.ipynb`. After obtaining the reaction embeddings of test set, I performed virtual screening on `clipzyme_screening_set.p`, excluding reactions that appeared in the training set. I also excluded enzymes from the training set when calculating the metrics, following the steps in `Results.ipynb`.

The results I obtained were as follows:

BEDROC85: 36.83%
BEDROC20: 57.23%
EF0.05: 13.13
EF0.1: 7.7

For this attempt, I have also attached my code for your reference.
[repro_attempt2.txt](https://github.com/user-attachments/files/17002455/repro_attempt2.txt)

Is there something I might have missed in my reproduction steps?

Thank you for your help!

**Comment by pgmikhael** [comment]
Hi,

Thank you for you reaching out! 

For training, there are two additional relevant arguments `train_esm_dir` and `pretrained_hub_dir` that are used by the `EnzymeReactionCLIP` model and the fair-esm model `FairEsm`. 

The results of excluding sequences in the training set is equivalent to Table 4 in the paper. The numbers you get are very similar to those (the first row of that table); not clear why they're not exactly the same, but they seem reasonable.

**Comment by ZhouGengmo** [comment]
Thank you for your response and assistance.

Regarding my reproduction attempt 2, the earlier description may be unclear.

Specifically, I used the metric calculation method provided in the `Analysis; Table 1: ablation` block from [analysis/Results.ipynb](https://github.com/pgmikhael/clipzyme/blob/clipzyme/analysis/Results.ipynb) in the repo. 

1. In this process, the reactions and the uniprot_id from the training set are extracted:
```
train_uniprots = set(d['uniprot_id'] for d in train_dataset.dataset)
train_reactions = set(d['reaction_string'] for d in train_dataset.dataset)
```

2. Then, reactions in the test set were filtered, excluding those that appeared in the training set. These reactions were not included in the metric calculations:
```
rxn_uni = test_dataset.dataset[ref_id]['reaction_string']
if rxn_uni in train_reactions:
    continue
```

3. Additionally, when calculating the metrics, any uniprot_id in the test set that also appeared in the training set was excluded:
```
label = [ (d['uniprot_id'] in train_uniprots) for d in data_samples]
print_metrics(metrics, percentiles, exclude_label=label)
```

In summary, both sequences and reactions from the training set were excluded during the calculations.

My concerns are:

1. To reproduce the results in the first row of Table 1, is it possible to use the released [clipzyme_model.zip](https://zenodo.org/records/11187895/files/clipzyme_model.zip?download=1) to infer reaction embeddings and perform virtual screening on the released `clipzyme_screening_set.p`, without training from scratch?

2. When obtaining the results in the first row of Table 1, were the two filtering steps mentioned above (2 and 3, excluding reactions and uniprot_id from the training set) applied, or were the metrics calculated directly without excluding these from the training set?

Thank you for your help!

**Comment by pgmikhael** [comment]
Hi,

1. Yes, you don't need to retrain.
2. Table 1 does not filter the screening set proteins (step 3 above). 

Also, except for paths, the notebook does not need to be changed (see the first for-loop under the `Analysis` section).

**Comment by ZhouGengmo** [comment]
Thank you so much for your help! I reproduce the results from the first row of Table 1 successfully by not filtering the screening set proteins.
I appreciate your support and assistance throughout this process.
